common/djangoapps/student/models.py

```diff
@@ -812,7 +812,16 @@ def user_post_save_callback(sender, **kwargs):
 
     changed_fields = user._changed_fields  # lint-amnesty, pylint: disable=protected-access
 
-    if 'is_active' in changed_fields or 'email' in changed_fields:
+    # DISABLE_AUTOENROLL_AFTER_COURSE_END flag is used to disable auto enrollment of user to a course
+    # if user is registering after course has ended
+    disabled_auto_enroll_after_course_end = settings.FEATURES.get('DISABLE_AUTOENROLL_AFTER_COURSE_END', False)
+
+    # Check if this is new user registration, or email change, or if
+    # DISABLE_AUTOENROLL_AFTER_COURSE_END flag is set to true
+    # The last condition is required to handle cases where the
+    # user registered after coursehas ended, but the course end date
+    # was later extended and the user logs in again.
+    if 'is_active' in changed_fields or 'email' in changed_fields or disabled_auto_enroll_after_course_end:
```

Comment:
How much overhead will adding `disabled_auto_enroll_after_course_end` introduce? We are going to query these enrollments on each user's login, account activation, and email/password change, right?

Reply:
I have now removed this, since the requirements are quite a bit simpler now and this does indeed introduce overhead, since it will query each time a user logs in.